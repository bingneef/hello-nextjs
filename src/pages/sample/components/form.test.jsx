import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

import Form from "./form.tsx";

describe("Form", () => {
  it("renders without error", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Form />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("matches the snapshot", () => {
    const component = renderer.create(<Form />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe("ui intereactions", () => {
    const component = renderer.create(<Form />);

    const rootInstance = component.root;
    let $counter = rootInstance.findByProps({ "test-id": "counter" });

    it("can increment the counter", async () => {
      expect($counter.props.children).toBe(0);

      const $increment = rootInstance.findByProps({ "test-id": "increment" });
      $increment.props.onClick();

      expect($counter.props.children).toBe(1);
    });

    it("can reset the counter", async () => {
      expect($counter.props.children).toBe(1);

      const $reset = rootInstance.findByProps({ "test-id": "reset" });
      $reset.props.onClick();

      expect($counter.props.children).toBe(0);
    });

    it("can decrement the counter", async () => {
      expect($counter.props.children).toBe(0);

      const $decrement = rootInstance.findByProps({ "test-id": "decrement" });
      $decrement.props.onClick();

      expect($counter.props.children).toBe(-1);
    });
  });
});
