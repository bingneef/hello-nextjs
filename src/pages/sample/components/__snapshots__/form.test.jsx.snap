// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form matches the snapshot 1`] = `
Array [
  <div
    test-id="counter"
  >
    0
  </div>,
  <button
    onClick={[Function]}
    test-id="increment"
  >
    +
  </button>,
  <button
    onClick={[Function]}
    test-id="decrement"
  >
    -
  </button>,
  <button
    onClick={[Function]}
    test-id="reset"
  >
    reset
  </button>,
]
`;
